ttnn/cpp/ttnn/operations/eltwise/binary/device/binary_device_operation.cpp

```diff
@@ -25,21 +30,24 @@ BinaryDeviceOperation::program_factory_t BinaryDeviceOperation::select_program_f
 
     if (height_a == height_b and width_a == width_b) {
         return ElementWiseMultiCore{};
-    } else if (height_b == 1 or width_b == 1) {
+    }
+    if (height_b == 1 or width_b == 1) {
         if (height_b == 1 and width_b == 1) {
             return BroadcastHeightAndWidthMultiCore{};
-        } else if (height_b == 1) {
-            if(tensor_args.input_tensor_a.is_sharded()){
-                if (tensor_args.input_tensor_a.get_legacy_shape()[0] == tensor_args.input_tensor_b.get_legacy_shape()[0]
-                        || tensor_args.input_tensor_a.get_legacy_shape()[0] > 1
-                        and tensor_args.input_tensor_b.get_legacy_shape()[0] == 1){
-                        return BroadcastHeightMultiCoreShardedOptimized{};
-                } else {
-                        return BroadcastHeightMultiCoreSharded{};
+        }
+        if (height_b == 1) {
+            if (tensor_args.input_tensor_a.is_sharded()) {
+                if (tensor_args.input_tensor_a.get_legacy_shape()[0] ==
```

Comment:
could you please change `get_legacy_shape` to `get_padded_shape`? Just to clearly represent the intention.